src/widgets/calendar/lv_calendar.c

```diff
@@ -134,17 +134,79 @@ void lv_calendar_set_showed_date(lv_obj_t * obj, uint32_t year, uint32_t month)
     uint8_t day_first = get_day_of_week(d.year, d.month, 1);
     uint8_t c;
     for(i = day_first, c = 1; i < act_mo_len + day_first; i++, c++) {
-        lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]), "%d", c);
+#if LV_USE_CALENDAR_LUNAR
+        if(calendar->use_lunar_calendar) {
+            lv_calendar_date_t lunar_temp;
+            lunar_temp = d;
+            lunar_temp.day = c;
+            const char * day_name = lv_calendar_get_day_name(&lunar_temp);
+            if(day_name != NULL)
+                lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]),
+                            "%d\n%s",
+                            c,
+                            day_name);
+            else
+                lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]), "%d", c);
+        }
+        else
+#endif
+            lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]), "%d", c);
     }
 
     uint8_t prev_mo_len = get_month_length(d.year, d.month - 1);
     for(i = 0, c = prev_mo_len - day_first + 1; i < day_first; i++, c++) {
-        lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]), "%d", c);
+#if LV_USE_CALENDAR_LUNAR
+        if(calendar->use_lunar_calendar) {
+            lv_calendar_date_t lunar_temp;
+            if(d.month == 1) {
+                lunar_temp.month = 12;
+                lunar_temp.year = d.year - 1;
+            }
+            else {
+                lunar_temp.month = d.month - 1;
+                lunar_temp.year = d.year;
+            }
+            lunar_temp.day = c;
+            const char * day_name = lv_calendar_get_day_name(&lunar_temp);
+            if(day_name != NULL)
+                lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]),
+                            "%d\n%s",
+                            c,
+                            day_name);
+            else
+                lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]), "%d", c);
+        }
+        else
+#endif
+            lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]), "%d", c);
         lv_buttonmatrix_set_button_ctrl(calendar->btnm, i + 7, LV_BUTTONMATRIX_CTRL_DISABLED);
     }
 
     for(i = day_first + act_mo_len, c = 1; i < 6 * 7; i++, c++) {
-        lv_snprintf(calendar->nums[i], sizeof(calendar->nums[0]), "%d", c);
+#if LV_USE_CALENDAR_LUNAR
+        if(calendar->use_lunar_calendar) {
+            lv_calendar_date_t lunar_temp;
+            if(d.month == 12) {
+                lunar_temp.month = 1;
+                lunar_temp.year = d.year + 1;
+            }
+            else {
+                lunar_temp.month = d.month + 1;
+                lunar_temp.year = d.year;
+            }
```

Comment:
Please also extract the repeated code to a function.
May `lunar_get_next_month` sound good?